nl_server/__init__.py

```diff
@@ -43,20 +43,34 @@ def create_app():
     log_level = logging.INFO
   logging.getLogger('werkzeug').setLevel(log_level)
 
-  app = Flask(__name__)
-  app.register_blueprint(routes.bp)
-
   # https://github.com/UKPLab/sentence-transformers/issues/1318
   if sys.version_info >= (3, 8) and sys.platform == "darwin":
     torch.set_num_threads(1)
 
+  app = Flask(__name__)
+  app.register_blueprint(routes.bp)
+
+  # Build the registry before creating the Flask app to make sure all resources
+  # are loaded.
   try:
-    loader.load_server_state(app)
+    reg = registry.build()
+
+    # Below is a safe check to ensure that the model and embedding is loaded.
+    # TODO: uncomment this code when fixing the crash issue during test.
+
+    # server_config = reg.server_config()
+    # idx_type = server_config.default_indexes[0]
+    # embeddings = reg.get_index(idx_type)
+    # query = server_config.indexes[idx_type].healthcheck_query
+    # result = search.search_vars([embeddings], [query]).get(query)
+    # if not result or not result.svs:
+    #   raise Exception(f'Registry does not have default index {idx_type}')
```

Comment:
this is causing crash during test. This is likely due to some threading issue during tests. comment out this for now and will come back to this next week.